### DISCUSSION TITLE
Looping Through OBX Segments

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#4855] [Q&A] [answered]
I saw an interesting option for looping through OBX segments. Can you tell me this works?
The item I'm wondering about is **for each (mOBX in msg..OBX)** 
```
var resultVal = new Array();
var counter = 0;

for each (mOBX in msg..OBX) 
{

        resultVal[counter] = mOBX['OBX.5']['OBX.5.1'].toString();
        counter += 1;
}

channelMap.put("resultVal",resultVal);```

**Comment by rjkroll** [comment]
Pretty easy to put that code into a test channel and see that it works, but unless you declared 'msg' somewhere, it probably won't work.

**Reply by pacmano1** [reply]
I don't understand the question being asked.   The forums has many references for looping over segments.   OP, what is "different" about the approach you asked about?

**Reply by flugruger** [reply]
pacmano1... thanks for responding. It just the msg..OBX format. I haven't seen it before. 
Like rjkroll says, I could build it and test... but I'm stuck in something and hoped the experts here would know.

**Reply by pacmano1** [reply]
There are perhaps hundreds of examples on the mirth forums that do a msg..<some segment name> to demonstrate iteration.

**Comment by pacmano1** [comment, marked as answer]
Marking as answered, this is a standard way to iterate.
